meetings/2021-10-13.md

```diff
@@ -0,0 +1,123 @@
Will address Larry’s request to have newer clang as a reason to bring bits of Conan into the mainline, try to make it so you can use a script to install any clang you want. Larry: don’t mind if there are extra steps, some of the years had several versions, some only had one. Aloys: may have made mistake, adding clang version was quite painful, makes things more complex than they should. Would like to go back to simpler, one release per year, and use Conan to install more / different things. Larry: because of runner time limits, can’t build a clang from scratch, so has to be binary install. Aloys: want to add Conan in separate folder in base CI image, so you can run a script to download any additional package you need. Will try with clang first, should be achievable before next meeting. Otherwise it will be 2022 to have a complete Conan transition, clang is a good motivation to do it, large package that adds lots of complexity to current images, so a good way to bring value from the work done so far. Will describe how it works at the next meeting / could do a demo. JF: can these script conversions be outsourced to other contributors? Aloys: it can be tricky to ask people to work in a branch they can’t use, so merging the work without changing the current system, just an additional layer of functionality, then it can be documented and others can contribute Conan recipes for additional packages. So will merge the new feature with documentation and presentation to this group, and won’t have to do all the packages myself. Finished IMath / OpenEXR last weekend, was tricky since Imath is a dummy package for OpenEXR 2 (went back to version 2.3 from 2019), lots of CMake bug workarounds required in Conan recipes, seems like a waste of time to focus on old versions, works much better in OpenEXR 3 and modern CMake. Not clear it’s worth making OpenEXR 2.3 work with the new system. JF: we should advertise to the project when Aloys is going to present. Patrick: is Conan the package manager we will promote, or just for CI? Aloys: wouldn’t got as far as that, but will be the one you can use to add more packages to your CI. Would like to make those recipes work on Windows (and eventually macOS), a lot easier on those OSes. Hoping we can contribute more OSes to existing packages. Conan has all the features needed for the aswf-docker project, supports multiple versions, looked at many package managers and seemed to be the only one that had all the features required. Do we promote it as the "ASWF Package Manager?" Good question. Patrick: if the project itself support Conan, for instance OCIO downloads and compiles third party libraries. Aloys: for each package, would make more sense to continue using CMake, some vendors use Conan (Foundry for instance building all their external libraries with Conan), so maybe we can help push Conan as the defacto package manager. But CMake is a known standard. JF: could Foundry be convinced to contribute some of those Conan recipes?
```

Comment:
I'm impressed you managed to capture all this from my babbles :) well done!

Reply:
And thanks for reading it!